llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -316,11 +321,20 @@ bool InterBlockScheduling::updateFixPoint(BlockState &BS) {
     // Iterate on CurMBB
     return false;
   }
-  if (!latencyConverged(BS)) {
-    BS.FixPoint.LatencyMargin++;
+
+  if (MachineInstr *MINeedsHigherCap = latencyConverged(BS)) {
+    auto Res = BS.FixPoint.PerMILatencyMargin.try_emplace(MINeedsHigherCap, 0);
+    if (BS.FixPoint.NumIters > MaxExpensiveIterations) {
+      // Increase the latency margin per instruction, unless we already iterated
+      // more than MaxExpensiveIterations without converging.
```

Comment:
nit: I think the comment should be on the outside of this if. And then perhaps also order the branches accordingly.